text/0000-float-semantics.md

```diff
@@ -0,0 +1,259 @@
+
+If you *do* use these operations on NaNs, then the exact behavior you see can depend on compiler version, compiler flags, target architecture, and it can even be non-deterministic (i.e., running the same operation on the same inputs twice can yield different results).
+The results produced in these cases do *not* always conform to the IEEE specification.
+See the reference section for what exactly is guaranteed.
+
+When a floating-point value is just passed around, its contents (including the bits of a NaN) do *not* change.
+
+When you use a floating-point operation in [`const` context](https://doc.rust-lang.org/reference/const_eval.html#const-context), the same specification applies: NaN bit patterns are non-deterministic.
+In particular, the bit pattern produced at compile-time can differ from the bit pattern produced by the same operation at run-time.
+
+Certain targets unfortunately are known to not implement these semantics precisely (see [below](#target-specific-problems)).
+The [platform support page](https://doc.rust-lang.org/rustc/platform-support.html) will list those caveats.
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This RFC specifies the behavior of `+`, `-` (unary and binary), `*`, `/`, `%`, `abs`, `copysign`, `mul_add`, `sqrt`, `as`-casts that involve floating-point types, and all comparison operations on floating-point types.
+Here, "floating-point types" are `f32` and `f64` and all similar types that might be added in the future such as `f16`, `f128`.
+Except for the cases handled below, these operations produce results that exactly match IEEE 754-2008 (with roundTiesToEven and default exception handling, without abruptUnderflow/flush-to-zero).
+`%` matches the behavior of `fmod` in C (this operation is not in the IEEE spec).
+When a floating-point value is just passed around (i.e., outside the operations above or any library-provided float operation), its representation bits do *not* change.
+
+Exceptions apply when the output of an operation is a NaN, and the operation is not a "bitwise" operation (unary `-`, `abs`, `copysign`).
+In that case, we generally follow [the same rules as LLVM](https://llvm.org/docs/LangRef.html#behavior-of-floating-point-nan-values).
+
+To be concrete, we first establish some terminology:
+A floating-point NaN value consists of a sign bit, a quiet/signaling bit, and a payload (which makes up the rest of the mantissa except for the quiet/signaling bit). Rust assumes that the quiet/signaling bit being set to ``1`` indicates a quiet NaN (QNaN), and a value of ``0`` indicates a signaling NaN (SNaN). In the following we will hence just call it the "quiet bit".
```

Comment:
Nit: mantissa -> significand

Reply:
Oh that's weird terminology, but Wikipedia agrees...